Gems/PhysX/Code/Editor/Source/MaterialEditor/PhysXMaterialEditorSystemComponent.cpp

```diff
@@ -0,0 +1,63 @@
+
+#include <AzCore/Serialization/SerializeContext.h>
+
+#include <Editor/Source/MaterialEditor/PhysXMaterialEditorSystemComponent.h>
+
+#include <Editor/Source/MaterialEditor/Window/MaterialEditorWindow.h>
+
+namespace PhysX
+{
+    void PhysXMaterialEditorSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<PhysXMaterialEditorSystemComponent, AZ::Component>()
+                ->Version(0);
+        }
+    }
+
+    PhysXMaterialEditorSystemComponent::PhysXMaterialEditorSystemComponent()
+        : m_notifyRegisterViewsEventHandler([this]() { RegisterAtomWindow(); })
+    {
+    }
+
+    PhysXMaterialEditorSystemComponent::~PhysXMaterialEditorSystemComponent() = default;
+
+    void PhysXMaterialEditorSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("PhysXMaterialEditorService"));
+    }
+
+    void PhysXMaterialEditorSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("PhysXMaterialEditorService"));
+    }
+
+    void PhysXMaterialEditorSystemComponent::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("O3DEMaterialEditorService"));
+    }
+
+    void PhysXMaterialEditorSystemComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+    }
+
+    void PhysXMaterialEditorSystemComponent::Activate()
+    {
+        if (auto* o3deMaterialEditor = O3DEMaterialEditor::O3DEMaterialEditorInterface::Get())
+        {
+            o3deMaterialEditor->ConnectNotifyRegisterViewsEventHandler(m_notifyRegisterViewsEventHandler);
+        }
+    }
+
+    void PhysXMaterialEditorSystemComponent::Deactivate()
+    {
+        m_notifyRegisterViewsEventHandler.Disconnect();
+    }
+
+    void PhysXMaterialEditorSystemComponent::RegisterAtomWindow()
+    {
+        O3DEMaterialEditor::RegisterViewPane<MaterialEditorWindow>("PhysX Materials");
+    }
```

Comment:
This is PhysX connecting to listen to notifications from O3DE Material Editor and registering its own window MaterialEditorWindow class when it's receiving the event.